### DISCUSSION TITLE
ctx.request.fields is unavailable

### DISCUSSION THREAD

**Discussion by spicy-sauce** [strapi/strapi#7297] [Legacy Topics]
Hi there,
When uploading files, they can be reached by using ctx.request.files, but it doesn't seem that additional (non-file) fields are available using ctx.request.fields (or koa-body's ctx.request.body.fields for that matter).
Is this the case or am I missing something?

Thanks!

**Comment by lauriejim** [comment]
Hello! What is the request you did and what is the code you are using?
This will help us to help you.

**Reply by spicy-sauce** [reply]
Hi, 
Thanks for answering 👍 

Here's the request payload:
-----------------------------30386908188232913942078611660
Content-Disposition: form-data; name="list_name"

test
-----------------------------30386908188232913942078611660
Content-Disposition: form-data; name="file"; filename="some_other_name.csv"
Content-Type: application/vnd.ms-excel

First Name,Last Name,Email,address,city,county,state,zip,phone1,Phone,company_name
<Sensitive Data Here>
-----------------------------30386908188232913942078611660--

So, I'm trying to get the value of list_name parameter:
module.exports = {
  parse: async (ctx) => {
    const file = ctx.request.files.file;
    const listName = ctx.request.fields.list_name; 
...

But listName is undefined as ctx.request,fields (and ctx.request.body.fields) is also undefined. 

Thanks in advance.

**Reply by lauriejim** [reply]
Hello! Thank you for your code.

Here is the source code of the create function that can manage multipart form.
https://strapi.io/documentation/v3.x/concepts/controllers.html#collection-type
-> check the `create` function 

Here is the code you can use:

```js
const { parseMultipartData } = require('strapi-utils');
const { data, files } = parseMultipartData(ctx);
```

**Comment by spicy-sauce** [comment]
Thanks, that worked like a charm!

**Reply by lauriejim** [reply]
Yeah cool :)
